src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8140,130 +8154,198 @@ void checkWellKnown(WellKnownMember member)
+                                original.Parameters[1] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
+                            {
+                                diagnostics.Add(syntax, useSiteInfo);
+                                makeCall(syntax, receiver, method, out indexerOrSliceAccess, out argumentPlaceholders);
+                                lookupResult.Free();
+                                return true;
+                            }
                         }
                     }
                 }
+
+                indexerOrSliceAccess = null;
+                argumentPlaceholders = default;
+                lookupResult.Free();
+                return false;
             }
 
-            patternSymbol = null;
+            void makeCall(SyntaxNode syntax, BoundExpression receiver, MethodSymbol method,
+                out BoundExpression indexerOrSliceAccess, out ImmutableArray<BoundIndexOrRangeIndexerPatternValuePlaceholder> argumentPlaceholders)
+            {
+                var startArgumentPlaceholder = new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, Compilation.GetSpecialType(SpecialType.System_Int32)) { WasCompilerGenerated = true };
+                var lengthArgumentPlaceholder = new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, Compilation.GetSpecialType(SpecialType.System_Int32)) { WasCompilerGenerated = true };
+                argumentPlaceholders = ImmutableArray.Create(startArgumentPlaceholder, lengthArgumentPlaceholder);
+
+                var analyzedArguments = AnalyzedArguments.GetInstance();
+                analyzedArguments.Arguments.Add(startArgumentPlaceholder);
+                analyzedArguments.Arguments.Add(lengthArgumentPlaceholder);
+
+                var boundMethodGroup = new BoundMethodGroup(
+                    syntax, typeArgumentsOpt: default, method.Name, ImmutableArray.Create(method),
+                    method, lookupError: null, BoundMethodGroupFlags.None, functionType: null, receiver, LookupResultKind.Viable);
+
+                indexerOrSliceAccess = BindMethodGroupInvocation(syntax, syntax, method.Name, boundMethodGroup, analyzedArguments,
+                    diagnostics, queryClause: null, allowUnexpandedForm: false, anyApplicableCandidates: out bool _);
+
+                analyzedArguments.Free();
+            }
+        }
+
+        private bool TryBindLengthOrCount(
+            SyntaxNode syntax,
+            BoundExpression receiver,
+            out BoundExpression lengthOrCountAccess,
+            BindingDiagnosticBag diagnostics)
+        {
+            var lookupResult = LookupResult.GetInstance();
+
+            Debug.Assert(receiver.Type is not null);
+            if (TryLookupLengthOrCount(syntax, receiver.Type, lookupResult, out var lengthOrCountProperty, diagnostics))
+            {
+                diagnostics.ReportUseSite(lengthOrCountProperty, syntax);
```

Comment:
>diagnostics.ReportUseSite(lengthOrCountProperty, syntax); [](http://example.com/codeflow?start=16&length=57)

Where do we report this use-site info when a regular property access is bound?

Reply:
I am still interested in the answer. However, this particular line looks redundant because ```TryLookupLengthOrCount``` reports use-cite info too.

Reply:
For a regular property access, we report the use-side info from the property in `ResultSymbol` (Binder_symbol.cs line 2057)

```
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.MissingMetadataTypeSymbol.ErrorInfo.get() Line 76 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.MissingMetadataTypeSymbol.TopLevel.ErrorInfo.get() Line 319 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.ErrorTypeSymbol.GetUseSiteInfo() Line 82 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbol.DeriveUseSiteInfoFromType(ref Microsoft.CodeAnalysis.UseSiteInfo<Microsoft.CodeAnalysis.CSharp.Symbols.AssemblySymbol> result, Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol type) Line 1051 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbol.DeriveUseSiteInfoFromType(ref Microsoft.CodeAnalysis.UseSiteInfo<Microsoft.CodeAnalysis.CSharp.Symbols.AssemblySymbol> result, Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations type, Microsoft.CodeAnalysis.CSharp.Symbol.AllowedRequiredModifierType allowedRequiredModifierType) Line 1082 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.PropertySymbol.CalculateUseSiteDiagnostic(ref Microsoft.CodeAnalysis.UseSiteInfo<Microsoft.CodeAnalysis.CSharp.Symbols.AssemblySymbol> result) Line 370 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PEPropertySymbol.GetUseSiteInfo() Line 741 C#
> Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag.ReportUseSite(Microsoft.CodeAnalysis.CSharp.Symbol symbol, Microsoft.CodeAnalysis.Location location) Line 110 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag.ReportUseSite(Microsoft.CodeAnalysis.CSharp.Symbol symbol, Microsoft.CodeAnalysis.SyntaxNode node) Line 98 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.ReportUseSite(Microsoft.CodeAnalysis.CSharp.Symbol symbol, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics, Microsoft.CodeAnalysis.SyntaxNode node) Line 1544 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.ResultSymbol.__resultSymbol|1046_0(Microsoft.CodeAnalysis.CSharp.LookupResult result, string simpleName, int arity, Microsoft.CodeAnalysis.SyntaxNode where, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics, bool suppressUseSiteDiagnostics, out bool wasError, Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceOrTypeSymbol qualifierOpt, Microsoft.CodeAnalysis.CSharp.LookupOptions options) Line 2057 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.ResultSymbol(Microsoft.CodeAnalysis.CSharp.LookupResult result, string simpleName, int arity, Microsoft.CodeAnalysis.SyntaxNode where, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics, bool suppressUseSiteDiagnostics, out bool wasError, Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceOrTypeSymbol qualifierOpt, Microsoft.CodeAnalysis.CSharp.LookupOptions options) Line 1708 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.GetSymbolOrMethodOrPropertyGroup(Microsoft.CodeAnalysis.CSharp.LookupResult result, Microsoft.CodeAnalysis.SyntaxNode node, string plainName, int arity, Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.Symbol> methodOrPropertyGroup, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics, out bool wasError, Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceOrTypeSymbol qualifierOpt) Line 7362 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindMemberOfType(Microsoft.CodeAnalysis.SyntaxNode node, Microsoft.CodeAnalysis.SyntaxNode right, string plainName, int arity, bool indexed, Microsoft.CodeAnalysis.CSharp.BoundExpression left, Microsoft.CodeAnalysis.SeparatedSyntaxList<Microsoft.CodeAnalysis.CSharp.Syntax.TypeSyntax> typeArgumentsSyntax, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations> typeArgumentsWithAnnotations, Microsoft.CodeAnalysis.CSharp.LookupResult lookupResult, Microsoft.CodeAnalysis.CSharp.BoundMethodGroupFlags methodGroupFlags, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics) Line 6732 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindInstanceMemberAccess(Microsoft.CodeAnalysis.SyntaxNode node, Microsoft.CodeAnalysis.SyntaxNode right, Microsoft.CodeAnalysis.CSharp.BoundExpression boundLeft, string rightName, int rightArity, Microsoft.CodeAnalysis.SeparatedSyntaxList<Microsoft.CodeAnalysis.CSharp.Syntax.TypeSyntax> typeArgumentsSyntax, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations> typeArgumentsWithAnnotations, bool invoked, bool indexed, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics, bool searchExtensionMethodsIfNecessary) Line 6480 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindMemberAccessWithBoundLeft(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax node, Microsoft.CodeAnalysis.CSharp.BoundExpression boundLeft, Microsoft.CodeAnalysis.CSharp.Syntax.SimpleNameSyntax right, Microsoft.CodeAnalysis.SyntaxToken operatorToken, bool invoked, bool indexed, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics) Line 6241 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindMemberAccess(Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax node, bool invoked, bool indexed, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics) Line 5948 C#
  Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionInternal(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax node, Microsoft.CodeAnalysis.CSharp.BindingDiagnosticBag diagnostics, bool invoked, bool indexed) Line 592 C#

```